reflex-dom-core/src/Reflex/Dom/Prerender.hs

```diff
@@ -130,10 +132,72 @@ instance (Adjustable t m, PrerenderBaseConstraints js t m, ReflexHost t) => Prer
         insertBefore df =<< deleteToPrerenderEnd doc
     holdDyn a0 a'
 
-data NoJavaScript -- This type should never have a HasJS instance
+newtype UnrunnableT t m a = UnrunnableT { runUnrunnableT :: Void -> m a }
 
-instance (js ~ NoJavaScript, SupportsStaticDomBuilder t m) => Prerender js t (StaticDomBuilderT t m) where
-  type Client (StaticDomBuilderT t m) = HydrationDomBuilderT GhcjsDomSpace t m
```

Comment:
Using `UnrunnableT` instead of `HydrationDomBuilderT GhcjsDomSpace` here seems orthogonal to the constraints change - or am I missing something?